sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -742,12 +750,41 @@ object ScalaJSPluginInternal {
 
   // These settings will be filtered by the stage dummy tasks
   val scalaJSRunSettings = Seq(
+      scalaJSMainModuleInitializer := {
+        if (scalaJSUseMainModuleInitializer.value) {
+          mainClass.value.fold {
+            throw new MessageOnlyException(
+                "No main class class detected, but " +
+                "scalaJSUseMainModuleInitializer was set to true. If you do " +
+                "not want to derive a main module initializer from " +
+                "discovered main classes, add " +
+                "`scalaJSMainModuleInitializer := None`")
+          } { mainClass =>
+            Some(ModuleInitializer.mainMethod(mainClass, "main"))
+          }
+        } else {
+          None
+        }
```

Comment:
Now I know what I found weird here :) This should detect the main initializer regardless of `scalaJSUseMainModuleInitializer`. It can then become non-optional.

This makes much more sense IMO. The key returns the main module initializer. Regardless of whether we want to use it or not (just like `mainClass` returns the main class, even if you `runMain` or not use it at all).

Maybe to better align with sbt, we should map on `mainClass` instead and put the error message in the updated `scalaJSEntryPoints ++=`?